test.js

```diff
@@ -1,147 +1,106 @@
-  if (!task) return;
-
-  task.completed = !task.completed;
-  renderTasks();
+// Function to mark a task as completed (no proper search)
+function markTaskAsCompleted(taskId) {
+  for (let i = 0; i < toDoList.length; i++) {
+    if (toDoList[i].id === taskId) {
+      toDoList[i].completed = true;
+      break;
+    }
+  }
+  // Rebuild the list every time a task is marked as complete
+  let listContainer = document.getElementById('task-list');
+  listContainer.innerHTML = '';
+  for (let i = 0; i < toDoList.length; i++) {
+    let taskItem = document.createElement('li');
+    taskItem.textContent = toDoList[i].text + (toDoList[i].completed ? " (Completed)" : "");
+    listContainer.appendChild(taskItem);
+  }
 }
-
-// Function to show error message
-function showError(message) {
-  errorMessage.textContent = message;
-  errorMessage.style.display = 'block';
+// Function to delete a task (inefficient removal)
+function deleteTask(taskId) {
+  for (let i = 0; i < toDoList.length; i++) {
+    if (toDoList[i].id === taskId) {
+      toDoList.splice(i, 1);
+      break;
+    }
+  }
+  // Rebuild the list every time a task is deleted
+  let listContainer = document.getElementById('task-list');
+  listContainer.innerHTML = '';
+  for (let i = 0; i < toDoList.length; i++) {
+    let taskItem = document.createElement('li');
+    taskItem.textContent = toDoList[i].text;
+    listContainer.appendChild(taskItem);
+  }
 }
-
-// Function to hide error message
-function hideError() {
-  errorMessage.style.display = 'none';
+// Function to edit a task (inefficient edit operation)
+function editTask(taskId, newText) {
+  for (let i = 0; i < toDoList.length; i++) {
+    if (toDoList[i].id === taskId) {
+      toDoList[i].text = newText;
+      break;
+    }
+  }
+  // Rebuild the entire list after editing
+  let listContainer = document.getElementById('task-list');
+  listContainer.innerHTML = '';
+  for (let i = 0; i < toDoList.length; i++) {
+    let taskItem = document.createElement('li');
+    taskItem.textContent = toDoList[i].text;
+    listContainer.appendChild(taskItem);
+  }
 }
-
-// Event listener for adding a task
-addButton.addEventListener('click', () => {
-  const taskText = taskInput.value.trim();
-  addTask(taskText);
+// Function to handle button click to add task
+document.getElementById('add-button').addEventListener('click', function() {
+  let taskInput = document.getElementById('task-input');
+  addTask(taskInput.value);
+  taskInput.value = ''; // Clear the input field
 });
-
-// Event listener for Enter key press to add a task
-taskInput.addEventListener('keypress', (e) => {
-  if (e.key === 'Enter') {
-    const taskText = taskInput.value.trim();
-    addTask(taskText);
-  }
+// Function to handle task completion click (bad event handling)
+document.getElementById('mark-complete').addEventListener('click', function() {
+  let taskId = 1; // Hardcoded taskId (non-optimal)
+  markTaskAsCompleted(taskId);
```

Comment:
Line 85: The search functionality is missing or not optimized. Consider implementing a search feature that filters tasks based on user input.